lib/sparrow/h2_worker.ex

```diff
@@ -414,6 +437,39 @@ defmodule Sparrow.H2Worker do
     :ok
   end
 
+  defp start_conn_backoff(config, :error, _delay_stream) do
+    case start_conn(config) do
+      {:ok, state} ->
+        {:ok, state}
+
+      {:error, reason} ->
+        _ =
+          Logger.error(
+            "action=starting_connection, domain=#{inspect(config.domain)}, port=#{
+              inspect(config.port)
+            }, tls_options=#{inspect(config.tls_options)}, maximum delay reached"
+          )
+
+        {:error, reason}
+    end
+  end
+
+  defp start_conn_backoff(config, {:ok, delay}, delay_stream) do
+    :timer.sleep(delay)
+
+    case start_conn(config) do
+      {:ok, state} ->
+        {:ok, state}
+
+      {:error, reason} ->
+        new_delay = Enum.fetch(delay_stream, 0)
```

Comment:
We probably still want to log something in this case. Maybe not an `error` but warning, or even info about failed attempt and retry would be nice to hav.e